util/verible-format.py

```diff
@@ -0,0 +1,196 @@
+
+            if len(diff) > 0 and show_cst:
+                cst = subprocess.Popen(['verible-verilog-syntax',
+                                        '--printtree', filename],
+                                       stdout=subprocess.PIPE,
+                                       stderr=subprocess.PIPE)
+                stdout, stderr = cst.communicate()
+                logger.info(stdout.decode())
+
+    return ret
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Format source code with Verible formatter')
+    parser.add_argument('-q', '--quiet', action='store_true', default=False,
+                        help='print only errors and warnings')
+    parser.add_argument('-v', '--verbose', action='store_true', default=False,
+                        help='print extra debug messages')
+    parser.add_argument('--show-diff', action='store_true', default=False,
+                        help='print diff (when files differs)')
+    parser.add_argument('--show-cst', action='store_true', default=False,
+                        help='print CST tree')
+    parser.add_argument('--progress', action='store_true', default=False,
+                        help='show progress')
+    parser.add_argument('--inplace', action='store_true', default=False,
+                        help='format files inplace (overwrite original files)')
+    parser.add_argument('-l', '--allowlist', action='store_true', default=False,
+                        help='process only files from allow list')
+    parser.add_argument('-a', '--all', action='store_true', default=False,
+                        help='process all files in repository (.sv and .svh)')
+    parser.add_argument('-f', '--files', metavar='file', type=str, nargs='+',
+                        help='process provided files')
+    args = parser.parse_args()
+
+    if not args.quiet:
+        logger.addHandler(logging.StreamHandler())
+        logger.setLevel(logging.INFO)
+
+        if args.verbose:
+            logger.setLevel(logging.DEBUG)
+
+    if args.inplace:
+        logger.warning('WARNING: Operating in-place - so make sure to make a backup or '
+                       'run this on an experimental branch')
+
+    if not is_verible_formatter_installed():
+        logger.error('verible-verilog-format either not installed or not visible in PATH')
+        sys.exit(1)
+    else:
+        logger.info('Using Verible formatter version: ' + get_verible_version())
+
+    logger.debug('repo_top: ' + get_repo_top())
+    logger.debug('verible exec: ' + str(get_verible_executable_path()))
+    logger.debug('verible version: ' + get_verible_version())
+    logger.debug('verible args: ' + str(VERIBLE_ARGS))
+
+    repo_top_dir = get_repo_top()
+
+    verible_files = []
+
+    if args.files:
+        for f in args.files:
+            filename_abs = repo_top_dir + '/' + f
+            if os.path.exists(filename_abs):
+                verible_files.append((f, filename_abs))
```

Comment:
This doesn't need the `if`: if `args.files` is empty, the loop won't run anyway.

Reply:
I think it does: `TypeError: 'NoneType' object is not iterable`

Reply:
Ah! You'll also need to add `default=[]` to the call to `parser.add_argument` where you add the `--files` argument (sorry, I didn't realise the type).

Reply:
Thanks! It works now.